includes/wf_crm_webform_base.inc

```diff
@@ -39,11 +43,21 @@ abstract class wf_crm_webform_base {
         return $this->_payment_processor;
       
       case 'contribution_page':
-        $contribution_page_id = wf_crm_aval($this->data, 'contribution:1:contribution:1:contribution_page_id');
-        if ($contribution_page_id && !$this->_contribution_page) {
-          $this->_contribution_page = wf_civicrm_api('contribution_page', 'getsingle', array('id' => $contribution_page_id));
-        }
         return $this->_contribution_page;
+        
+      case 'tax_rate':
+        $taxSettings = CRM_Core_BAO_Setting::getItem(CRM_Core_BAO_Setting::CONTRIBUTE_PREFERENCES_NAME, 'contribution_invoice_settings');
+        if(isset($taxSettings['invoicing'])){
+          if($taxSettings['invoicing']){
+            if ($this->_contribution_page) {
```

Comment:
Oh I see, your workaround was to move the code that fetched contribution page out of its condition.
Assuming it doesn't cause any kind of weird recursion with a magic method being called from within another magic method, I think my suggestion might be cleaner.